### ISSUE TITLE
[Android] 4.0.1.305340-nightly breaks ListView recylcing

### ISSUE THREAD

**Issue by adrianknight89** [xamarin/Xamarin.Forms#5838]
I have a `DataTemplate` with a `ViewCell` whose content is a `Grid`. The grid has multiple auto-sized rows. One of the rows is sized according to view model bindings, but Android does not respect these bindings if I scroll up and down on the `ListView`. So, it looks like cell recycling is broken.

```
<forms:CachedImage Grid.Row="2" Grid.Column="0" Grid.ColumnSpan="3" Style="{x:StaticResource ImageStyle}">
     <forms:CachedImage.GestureRecognizers>
          <TapGestureRecognizer Tapped="ImageTapGestureRecognizer_Tapped"/>
     </forms:CachedImage.GestureRecognizers>
</forms:CachedImage>
```

The style sets `WidthRequest` and `HeightRequest` of the image according to the view model. Reverting to 4.0.1.268567-nightly fixes this issue.

Another issue I see is the `TapGestureRecognizer` command is no longer firing when I x:Reference it to the parent view's view model, and so I ended up wiring the `Tapped` event and calling the command from it (I have not verified yet if this is too due to the latest nightly build).

Let me know if you prefer to be notified of nightly errors through another channel instead of Github. Also, is there a way to see code delta between nightly builds? I don't think you're tagging these builds on here.

**Comment by samhouts** [comment]
Hi @adrianknight89 ! Nope, this is the perfect place to report these issue, thanks! 

 268567 was built from https://github.com/xamarin/Xamarin.Forms/commit/ce148bf4c81f7d1da5ef74c7a2a43cca6ca73331. 305340 was build from https://github.com/xamarin/Xamarin.Forms/commit/740654ff5654b6f572ddb6be7b34991bb2d73d02. If you have a sample project, we can help investigate. Thanks!!

**Comment by adrianknight89** [comment]
@samhouts Thanks. I'll try to create a repro tonight. Just to confirm, the nightly builds are based off of `master`, correct?

**Comment by samhouts** [comment]
That's correct. Thanks, @adrianknight89 !

**Comment by adrianknight89** [comment]
@samhouts Here's a simple repro:

[TestN.zip](https://github.com/xamarin/Xamarin.Forms/files/3056969/TestN.zip)

**Comment by InquisitorJax** [comment]
Here's another sample project
https://github.com/InquisitorJax/Xamarin-Forms-4-Layouts

**Comment by adrianknight89** [comment]
still exists in 4.0.1.383393-nightly

**Comment by rmarinho** [comment]
I think is this PR , i didn't found a fix yet, 
 https://github.com/xamarin/Xamarin.Forms/commit/ee01ce0cf6050ec1c1d5ca39ae1e38add5c97f13

**Comment by adrianknight89** [comment]
@paymicro Do you have any insight on this? I think you've been making changes to layout stuff recently. #6390

**Comment by paymicro** [comment]
Hi @adrianknight89 Layout changes in PR #6390 should be fixes this issue.

**Comment by adrianknight89** [comment]
That's what I was hoping to hear, hence I mentioned that PR before Rui digs deeper into the issue. Thank you!

**Comment by samhouts** [comment]
closed by #6390
